Source/mpqapi.cpp

```diff
@@ -312,7 +312,7 @@ void ByteSwapHdr(_FILEHEADER *hdr)
 void InitDefaultMpqHeader(Archive *archive, _FILEHEADER *hdr)
 {
 	std::memset(hdr, 0, sizeof(*hdr));
-	hdr->signature = LOAD_LE32("MPQ\x1A");
+	hdr->signature = '\x1AQPM';

```

Comment:
Above (at line 265) `LOAD_LE32("MPQ\x1A");` was replaced with `SDL_SwapLE32('\x1AQPM');`.

On this line, the same code was replaced with `'\x1AQPM'`, without a call to `SDL_SwapLE32`. Was this difference intentional? Same question for line 325.

Reply:
that part was a temporary revert to get things back in to working order on BE, but this is exactly the inconsistency that i'm trying to find the proper solution for.